packages/core/src/testInteg/perf/utilities.ts

```diff
@@ -0,0 +1,49 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as sinon from 'sinon'
+import { FileSystem } from '../../shared/fs/fs'
+
+/**
+ * Provide an upper bound on total number of system calls done through our fs module.
+ * @param fsSpy filesystem spy (Ex. `sinon.spy(fs)`)
+ * @returns count of operations
+ */
+export function getFsCallsUpperBound(fsSpy: sinon.SinonSpiedInstance<FileSystem>): number {
+    return getFsReadsUpperBound(fsSpy) + getFsWritesUpperBound(fsSpy)
+}
+
+/**
+ * Provide an upper bound on the number of filesystem reads done through our fs module.
+ * This is an upper bound because some of the functions call eachother.
+ * @param fsSpy filesystem spy (Ex. `sinon.spy(fs)`)
+ * @returns value
+ */
+export function getFsReadsUpperBound(fsSpy: sinon.SinonSpiedInstance<FileSystem>): number {
+    return (
+        fsSpy.readFileBytes.callCount +
```

Comment:
Just a thought: do we actually need spies or could our fs.ts module just increment a count?